### ISSUE TITLE
Stack overflow when auto-mapping self referencing type

### ISSUE THREAD

**Issue by BalassaMarton** [hughbe/excel-mapper#36]
`AutoMapper.TryCreateClassMap` probably doesn't detect reference loops in the type.
Repro:
```cs
public class MyType {
    public MyType Child { get; set; }
}
```
```
AutoMapper.TryCreateClassMap<MyType>(FallbackStrategy.SetToDefaultValue, out var map);
```

**Comment by hughbe** [comment]
What would you expect? How should it be mapped automatically?

**Comment by BalassaMarton** [comment]
There could be some config option that controls this behavior (skip or throw). I ended up implementing my own `ExcelClassMap` that just skips anything that's not a trivial value type.

**Comment by hughbe** [comment]
In https://github.com/hughbe/excel-mapper/pull/48 I added `ExcelIgnoredAttribute` which basically marks the property/field as ignored for the purposes of auto-mapping.

I added basic validation for recursive structures, limited to one-level (i.e. we don't catch diamond recursive structures but I didn't really want to go into that). We now throw ExcelMappingException with an error message suggesting that you add `ExcelIgnore`
